### PR TITLE
Create API endpoints to create bookings

### PR DESCRIPTION
**Pull request by sidsbrmnn**
Create Next API endpoints at `/api/bookings` to initialize and verify a payment
session using Stripe.

To generate a new Payment Intent or a Subscription,

```http
POST /api/bookings/initialize
Content-Type: application/json

{
  "bathrooms": 1,
  "bedrooms": 1,
  "cleaningType": "Standard",
  "interval": "One Time"
}
```

To verify the payment and confirm the booking,

```http
POST /api/bookings/verify
Content-Type: application/json

{
  "interval": "One Time",
  "paymentId": "pi_1DpcVL2eZvKYlo2CQT4lFQr1", // use subscriptionId in case of subscriptions
  "bathrooms": 1,
  "bedrooms": 1,
  "cleaningType": "Standard"
}
```

- Install stripe and next-connect
- Create stripe helpers
- Added support for interval
- Added types for Props
- Update stripe helpers with Interval
- Create POST /bookings/initialize route
- Update calculate total price with interval
- Create POST /bookings/verify to verify payments
- Create .env.example
- Create .editorconfig

### CODE REVIEW

File: src/lib/stripe-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+export function formatAmountForStripe(amount: number): number {
+  let numberFormat = new Intl.NumberFormat(["en-US"], {
+    style: "currency",
+    currency: "USD",
+    currencyDisplay: "symbol",
+  });
+  const parts = numberFormat.formatToParts(amount);
+  let zeroDecimalCurrency: boolean = true;
+  for (let part of parts) {
```
Comment (AdithyaBhat17) [inline src/lib/stripe-helpers.ts:39]: Could use const here.

File: src/pages/api/bookings/initialize.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    const subscription = await stripe.subscriptions.create({
+      customer: "cus_KVfi2rxCMiKJ4v",
+      items: [
+        {
+          price_data: {
+            currency: "usd",
+            product: products[req.body.interval].id,
+            unit_amount: amount,
+            recurring: getRecurringPrice(req.body.interval),
+            tax_behavior: "inclusive",
```
Comment (AdithyaBhat17) [inline src/pages/api/bookings/initialize.ts:59]: How does this work? At Homero, we already compute the tax and add it to the total price so I don't think we need stripe to charge tax rates as well.

File: src/lib/stripe-helpers.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+// Create type Interval for weekly, biweekly, monthly
+export type Interval = "Weekly" | "Every 2 weeks" | "Every 3 weeks";
+
+// Get recurring price data from interval
+export const getRecurringPrice = (
+  interval: Interval
+): Stripe.SubscriptionCreateParams.Item.PriceData.Recurring => {
+  switch (interval) {
+    case "Weekly":
+      return { interval: "week", interval_count: 1 };
```
Comment (AdithyaBhat17) [inline src/lib/stripe-helpers.ts:12]: You could set the interval as week outside the switch case and just change interval count based on the case.

**Review by AdithyaBhat17 — APPROVED** [review]

File: src/lib/stripe-helpers.ts (hunk 3 above)
Comment (AdithyaBhat17) [reply src/lib/stripe-helpers.ts:12]: Or create an object with intervals as keys and skip using switch cases.

File: src/pages/api/bookings/initialize.ts (hunk 2 above)
Comment (sidsbrmnn) [reply src/pages/api/bookings/initialize.ts:59]: "inclusive" states that the rates we compute includes the taxes. So that Stripe doesn't add more

**Pull request merged**